wdl-analysis/src/analyzer.rs

```diff
@@ -409,11 +414,19 @@ where
                 bail!("`{path}` is a file, not a directory", path = path.display());
             }
 
-            for result in WalkDir::new(&path).follow_links(true) {
-                let entry = result.with_context(|| {
-                    format!("failed to read directory `{path}`", path = path.display())
-                })?;
-                if !entry.file_type().is_file()
+            let mut walker = WalkBuilder::new(&path);
+            if let Some(ignore_filename) = config.ignore_filename() {
+                walker.add_custom_ignore_filename(ignore_filename);
+            }
+            let walker = walker.standard_filters(false).parents(true).build();
+
+            for result in walker {
+                let entry = match result {
+                    Ok(entry) => entry,
+                    Err(_) => continue, // ignore any errors parsing the ignore files
+                };
+
+                if !entry.file_type().is_some_and(|e| e.is_file())
                     || entry.path().extension().and_then(OsStr::to_str) != Some("wdl")
                 {
                     continue;
```

Comment:
Whoa, TIL about `Option::is_some_and()`, that's a neat function! Between that, the negation, and the `||`ing with another condition, though, I think it is difficult to read and understand the intent here. It also made it easy to miss that an entry without a file type would actually fall through and attempt to get analyzed, probably exploding at some later point.

Here's how I would rewrite this. It's more verbose and is somewhat repetitive, but imo requires less mental overhead to understand.

```suggestion
                // Skip entries without a file type
                let Some(file_type) = entry.file_type() else {
                    continue;
                };
                // Skip non-files
                if !file_type.is_file() {
                    continue;
                }
                // Skip files without a `.wdl` extension
                if entry.path().extension() != Some(&OsStr::new("wdl")) {
                    continue;
```